executors/cpp/message_fmt2.cpp

```diff
@@ -0,0 +1,180 @@
+        params_name = json_object_get_string(params_name_obj);
+        u_params_name = params_name.c_str();
+      }
+      json_object *params_value_obj =
+          json_object_object_get(param_obj, "value");
+      if (params_value_obj) {
+        params_value = json_object_get_string(params_value_obj);
+        u_params_value = params_value.c_str();
+      }
+      argsBuilder[u_params_name] = Formattable(u_params_value);
+    }
+  }
+
+  // Start filling the return data.
+  json_object *return_json = json_object_new_object();
+  json_object_object_add(return_json, "label", label_obj);
+
+  bool no_error = true;
+
+  if (U_FAILURE(errorCode)) {
+    const char* error_name = u_errorName(errorCode);
+    json_object_object_add(
+        return_json,
+        "error", json_object_new_string("error in constructor"));
+    json_object_object_add(
+        return_json,
+        "error_detail", json_object_new_string(error_name));
+    no_error = false;
+  }
+
+  UParseError parseError;
+
+  MessageFormatter::Builder builder(errorCode);
+  MessageFormatter mf = builder.setPattern(u_src, parseError, errorCode)
+                        .setLocale(displayLocale)
+                        .build(errorCode);
+
+  if (U_FAILURE(errorCode)) {
+    const char* error_name = u_errorName(errorCode);
+    json_object_object_add(
+        return_json,
+        "error", json_object_new_string("error in builder.setPattern"));
+    json_object_object_add(
+        return_json,
+        "error_detail", json_object_new_string(error_name));
+    return json_object_to_json_string(return_json);
+  }
+
+  MessageArguments args(argsBuilder, errorCode);
+  if (U_FAILURE(errorCode)) {
+    const char* error_name = u_errorName(errorCode);
+    json_object_object_add(
+        return_json,
+        "error", json_object_new_string("error in constructing args"));
+    json_object_object_add(
+        return_json,
+        "error_detail", json_object_new_string(error_name));
+    return json_object_to_json_string(return_json);
+  }
+
+  UnicodeString result_ustring = mf.formatToString(args, errorCode);
+  if (U_FAILURE(errorCode)) {
+    const char* error_name = u_errorName(errorCode);
+    json_object_object_add(
+        return_json,
+        "error", json_object_new_string("error in formatToString"));
+    json_object_object_add(
+        return_json,
+        "error_detail", json_object_new_string(error_name));
+    return json_object_to_json_string(return_json);
+  }
+
+  int32_t chars_out;  // Extracted characters from Unicode string
+  char test_result[1000] = "";
+
+  // Get the resulting value as a string
+  result_ustring.extract(test_result, 1000, nullptr, errorCode);
+
+  if (U_FAILURE(errorCode)) {
+  } else {
+    // It worked!
+    json_object_object_add(return_json,
+                           "result",
+                           json_object_new_string(test_result));
+  }
+
+  return json_object_to_json_string(return_json);
+}
+#endif
```

Comment:
nit: add a newline (so that the `#endif` doesn't blend into the background) and a comment indicating the condition of the if statement, since it encapsulates so much code
```suggestion
}

#endif  // U_ICU_VERSION_MAJOR_NUM >= 75
```

Reply:
OK